usersession/userd/settings.go

```diff
@@ -77,15 +94,116 @@ func allowedSetting(setting string) bool {
 	return snap.ValidAppName(base)
 }
 
+func settingWithSub(setting string, subproperty string) string {
+	return fmt.Sprintf("%s/%s", setting, subproperty)
+}
+
+func settingForLog(setting string) string {
+	settingParts := strings.SplitN(setting, "/", 2)
```

Comment:
The splitting here and elsewhere of what is joined in settingWithSub probably shows that we shouldn't use a string anymore for setting but some kind of struct. We can address that in a follow up though.